tests/basic/storage_pool_test.go

```diff
@@ -9993,3 +9994,283 @@ func exitPoolMaintenance(poolId string) {
+			deletePoolAndValidateFaliure(invalidPoolID, selectedNode, errRegExp)
+		}
+
+		errRegExp = regexp.MustCompile("unknown shorthand flag: '1' in -1")
+		deletePoolAndValidateFaliure("-1", selectedNode, errRegExp)
+
+		log.FailOnError(ExitPoolMaintenance(*selectedNode), "failed to exit pool maintenance mode")
+	})
+
+	ItLog = "Delete pool that has a volume on it"
+	It(ItLog, func() {
+		log.InfoD(stepLog)
+
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("pooldeletewithvol-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		selectedPoolUUID := pickPoolToResize()
+		log.InfoD("Pool UUID on which IO is running [%s]", selectedPoolUUID)
+		selectedPool, err := GetStoragePoolByUUID(selectedPoolUUID)
+		log.FailOnError(err, "error getting storage pool with UUID [%s]", selectedPoolUUID)
+		selectedNode, err := GetNodeWithGivenPoolID(selectedPoolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", selectedPoolUUID)
+
+		log.FailOnError(EnterPoolMaintenance(*selectedNode), "failed to enter pool maintenance mode")
+		errRegExp := regexp.MustCompile("Cannot delete pool: Following volumes have data on pool")
+		deletePoolAndValidateFaliure(fmt.Sprintf("%v", selectedPool.ID), selectedNode, errRegExp)
+		log.FailOnError(ExitPoolMaintenance(*selectedNode), "failed to exit pool maintenance mode")
+	})
+
+	ItLog = "Delete pool while a volume on the pool is in resync"
+	It(ItLog, func() {
+		log.InfoD(stepLog)
+		// don't schedule the appList so that any pool we pick for repl-add
+		// and delete doesn't have any other volumes
+
+		// schedule the fio app
+		appList := Inst().AppList
+		defer func() {
+			Inst().AppList = appList
+		}()
+		Inst().AppList = []string{"fio-storagepool"}
+		log.InfoD("scheduling apps ")
+		appNamespace := fmt.Sprintf("pooldeletewithresync-%s", Inst().InstanceID)
+		contexts := ScheduleApplicationsOnNamespace(appNamespace, "pooldeletewithresync")
+		defer appsValidateAndDestroy(contexts)
+
+		// waiting for the data to be written before performing ha-update
+		testVolume, err := GetVolumeWithMinimumSize(contexts, 50)
+		log.FailOnError(err, "error selecting volumes for resync")
+
+		contextAppKey := contexts[0].App.Key
+
+		// find pool to repl add and then delete
+		replAddNode, err := findNodeForReplAdd(testVolume)
+		log.FailOnError(err, "failed to find node for repl add")
+		dash.VerifyFatal(replAddNode != nil, true, "expect to find a node to do repl add")
+		replAddPool := replAddNode.StoragePools[0]
+		// TODO add check that pool doesn't contain metadata
+
+		stepLog = fmt.Sprintf("Increasing the volume repls for %s", contextAppKey)
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			log.Infof("Increase repl for volume: %v", testVolume.ID)
+			pxctlCmdFull := fmt.Sprintf("v ha-update -r 3 --node %v %v", replAddPool.Uuid, testVolume.ID)
```

Comment:
how do you confirm that the volume to be ha-updated has repl=2?

Reply:
We don't care about that for this test. The volume enters resync is all we are about.